test/integration/backward_compatible/HazelcastClientTest.js

```diff
@@ -116,44 +116,34 @@ configParams.forEach(function (cfg) {
             expect(info.labels).to.deep.equal(new Set());
         });
 
-        it('getDistributedObjects returns all dist objects', function (done) {
+        it('getDistributedObjects returns all dist objects', function () {
             managed.getObject(client.getMap.bind(client, 'map'));
             managed.getObject(client.getSet.bind(client, 'set'));
-            setTimeout(function () {
-                client.getDistributedObjects().then(function (distObjects) {
-                    try {
-                        const names = distObjects.map((o) => {
-                            return o.getName();
-                        });
-                        expect(names).to.have.members(['map', 'set']);
-                        done();
-                    } catch (e) {
-                        done(e);
-                    }
+            TestUtil.assertTrueEventually(function () {
```

Comment:
I think you need to await the promise returned by the ` TestUtil.assertTrueEventually`. 

Also, you can make the callback an async function and await the `client.getDistributedObjects` call which would make it much more readable.

The same applies to other places

Reply:
Thanks, good catch. In addition to fixing them, I adjusted the timeout to be 5 seconds. It should be enough.